ports/esp32/main.c

```diff
@@ -206,3 +220,9 @@ void *esp_native_code_commit(void *buf, size_t len, void *reloc) {
     memcpy(p, buf, len);
     return p;
 }
+
+#if CONFIG_IDF_TARGET_ESP32S3
+float __ieee754_sqrtf(float x) {
```

Comment:
@UnexpectedMaker why is this needed?  surely it needs a proper implementation :)

Reply:
That was what `we` added when we did that video call a while back as something in the compiler was looking for it and it was falling over without it.

Reply:
Oh, I don't remember that!  Do you have any S3 board definitions that compile?

Reply:
I'll need to test out my S3 board again today and I'll get back to you with files for a working S3 board.

Reply:
> Oh, I don't remember that! Do you have any S3 board definitions that compile?

I've not had a chance to look into this yet, sorry. Feel free to remove it for now as its S3 specific only.

Reply:
Ok, removed this for now.